### PR TITLE
Handle check url

### PR DESCRIPTION
**Pull request by gcode-de**

### CODE REVIEW

File: backend/src/test/java/de/samuelgesang/backend/sites/SiteControllerTest.java [hunk 1]
```diff
@@ -35,20 +35,22 @@ void setUp() {
...
+//        when(oauth2User.getAttributes()).thenReturn(attributes);
+//
+//        List<Site> sites = List.of(new Site("1", "Google", "https://google.com", "", "user123", "string", new ArrayList<>()));
+//        when(siteService.getAllSites("user123")).thenReturn(sites);
+//
+//        List<SiteWithCrawlsDTO> result = siteController.getAllSites(oauth2User);
+//
+//        assertEquals(1, result.size(), "The result size should be 1");
+//        assertEquals("Google", result.get(0).getName(), "The site name should be Google");
+//    }
```
Comment (josch87) [inline backend/src/test/java/de/samuelgesang/backend/sites/SiteControllerTest.java:53]: Do not comment out tests. Instead if you do not want to fix it right now use `@Disabled` annotation for JUnit and `.skip` method for Jest. This way you will get reminded that there are tests that you need to work on. In a big project you will forget about and never find this test again. 

```suggestion
    @Disabled
    @Test
    @WithMockUser
    void testGetAllSites() {
        Map<String, Object> attributes = new HashMap<>();
        attributes.put("sub", "user123");
        when(oauth2User.getAttributes()).thenReturn(attributes);

        List<Site> sites = List.of(new Site("1", "Google", "https://google.com", "", "user123", "string", new ArrayList<>()));
        when(siteService.getAllSites("user123")).thenReturn(sites);

        List<SiteWithCrawlsDTO> result = siteController.getAllSites(oauth2User);

        assertEquals(1, result.size(), "The result size should be 1");
        assertEquals("Google", result.get(0).getName(), "The site name should be Google");
    }
```

File: frontend/src/pages/Home.tsx [hunk 2]
```diff
@@ -66,8 +67,45 @@ const Home: React.FC<HomeProps> = ({sites, refreshSites}: HomeProps) => {
         setEditSiteId(null);
     };
 
-    const handleCheckUrl = (crawlId: string, url: string) => {
-        console.log("Toggle checkbox:", crawlId, url);
+    const handleCheckUrl = async (crawlId: string, url: string, newState: boolean) => {
+        console.log('Toggle checkbox:', crawlId, url, newState);
```
Comment (josch87) [inline frontend/src/pages/Home.tsx:71]: Found a console.log 😉

File: frontend/src/pages/Home.tsx [hunk 3]
```diff
@@ -66,8 +67,45 @@ const Home: React.FC<HomeProps> = ({sites, refreshSites}: HomeProps) => {
...
+
+        const payload = {
+            url: url,
+            checked: newState,
+        };
+
+        try {
+            const response = await axios.put(`/api/crawls/update-url-status/${crawlId}`, payload);
+            console.log('URL checked status updated successfully:', response.data);
```
Comment (josch87) [inline frontend/src/pages/Home.tsx:80]:
```suggestion
```

File: frontend/src/pages/Home.tsx [hunk 4]
```diff
@@ -66,8 +67,45 @@ const Home: React.FC<HomeProps> = ({sites, refreshSites}: HomeProps) => {
...
+                                    };
+                                }
+                                return crawl;
+                            }),
+                        };
+                    }
+                    return site;
+                })
+            );
+            console.log('Sites state updated:', sites); // Fügen Sie dies hinzu, um den aktualisierten State zu protokollieren
```
Comment (josch87) [inline frontend/src/pages/Home.tsx:103]:
```suggestion

```

**Review by josch87 — CHANGES_REQUESTED** [review]

File: backend/src/main/java/de/samuelgesang/backend/url_chunk/UrlObject.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+import lombok.*;
+
+@Getter
+@Setter
+@ToString
+public class UrlObject {
+    private String url;
+
+    public UrlObject(String url, boolean b) {
```
[the remark this answers is not in the document]
Comment (gcode-de) [reply backend/src/main/java/de/samuelgesang/backend/url_chunk/UrlObject.java:11]: oh, this file needs to be deleted

File: backend/src/test/java/de/samuelgesang/backend/sites/SiteControllerTest.java (hunk 1 above)
Comment (gcode-de) [reply backend/src/test/java/de/samuelgesang/backend/sites/SiteControllerTest.java:53]: nice, thanks!

File: backend/src/main/java/de/samuelgesang/backend/crawls/CrawlController.java [hunk 6]
```diff
@@ -60,4 +58,28 @@ public ResponseEntity<String> crawlSiteById(@PathVariable String siteId, @Authen
...
+                return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+            }
+            log.info("Received request to update URL checked status: crawlId={}, updateUrlStatusDTO={}", crawlId, updateUrlStatusDTO);
+
+            Crawl updatedCrawl = crawlService.updateUrlCheckedStatus(crawlId, updateUrlStatusDTO);
+
+            if (updatedCrawl != null) {
+                return ResponseEntity.ok(updatedCrawl);
+            } else {
+                return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
```
[the remark this answers is not in the document]
Comment (gcode-de) [reply backend/src/main/java/de/samuelgesang/backend/crawls/CrawlController.java:77]: nice! I will do this in the refactoring ticket

**Pull request merged**